mypy/plugin.py

```diff
@@ -332,3 +374,53 @@ def int_pow_callback(ctx: MethodContext) -> Type:
         else:
             return ctx.api.named_generic_type('builtins.float', [])
     return ctx.default_return_type
+
+
+def add_method(
+        info: TypeInfo,
+        method_name: str,
+        args: List[Argument],
+        ret_type: Type,
+        self_type: Type,
+        function_type: Instance) -> None:
+    from mypy.semanal import set_callable_name
+
+    first = [Argument(Var('self'), self_type, None, ARG_POS)]
+    args = first + args
+
+    arg_types = [arg.type_annotation for arg in args]
+    arg_names = [arg.variable.name() for arg in args]
+    arg_kinds = [arg.kind for arg in args]
+    assert None not in arg_types
+    signature = CallableType(arg_types, arg_kinds, arg_names,
+                             ret_type, function_type)
+    func = FuncDef(method_name, args, Block([]))
+    func.info = info
+    func.is_class = False
+    func.type = set_callable_name(signature, func)
+    func._fullname = info.fullname() + '.' + method_name
+    info.names[method_name] = SymbolTableNode(MDEF, func)
+
+
+def attr_s_callback(ctx: ClassDefContext) -> None:
+    """Add an __init__ method to classes decorated with attr.s."""
+    info = ctx.cls.info
+    # TODO: Handle default arguments
+    has_default = {}  # type: Dict[str, Expression]
+    args = []
+
+    for name, table in info.names.items():
+        if table.type:
```

Comment:
I think this should be treated differently. You should look at the right hand sides for assignments in the class and take those that have ``attr.ib`` (note however, that I am not an expert in ``attrs`` so I would like to see many tests for this reviewed by an ``attrs`` maintainer).

Reply:
If possible I'd like to remove all this code from the PR.  The actual plugin will require possibly more thought.  Is it weird to just have a commit that creates a plugin but doesn't actually use it?

Also it turns out I can't actually use a plugin unless attr.pyi exists.  So I'd have to figure out how to bring in https://github.com/python-attrs/attrs/pull/238